docs/v1/P4Runtime-Spec.mdk

```diff
@@ -4755,8 +4835,19 @@ controller populates the `MasterArbitrationUpdate` field in this message using
 its `role_id` and `election_id` and the `device_id` of the device, as explained
 in detail in the [Master-Slave Arbitration and Controller
 Replication](#sec-master-slave-arbitration-and-controller-replication)
-section. For any given (`device_id`, `role_id`), the controller with the highest
-`election_id` is the master and the rest are slaves.
+section. For any given `(device_id, role_id)`, the P4Runtime server keeps track
+of the highest `election_id` that it has ever received. If a controller's
+`election_id` is equal to the highest `election_id` that the server has ever
+received, that controller is the master. All other controllers are slaves. Note
+that it is possible that all controllers are slaves, and that there is no
+master. There can be at most one master, because for any given
+`(device_id, role_id)`, each connected controller has a unique `election_id`.
+
+To ensure that the invariant holds across reboots, the device needs to persist
```

Comment:
Yeah this was not right, if it is persisted across restarts then there is no way to reset the election ID. I clarified that all switches have "full restarts" (where the forwarding state is cleared etc), and then the election_id is also reset (and NOT persisted).  Some targets may also support in-service software upgrade (ISSU), ie. restart while forwarding traffic.  In that case, the election_id must be persisted.

Does that satisfy your "nice-to-have" wish, since ISSU is optional?